### PR TITLE
Fix exception causes all over the codebase

### PR DESCRIPTION
**Pull request by cool-RR**
@birkenfeld This is a continuation of #1478

### CODE REVIEW

File: pygments/filters/__init__.py [hunk 1]
```diff
@@ -777,8 +777,10 @@ def __init__(self, **options):
             # issubclass() will raise TypeError if first argument is not a class
             if not issubclass(self.exception, Exception):
                 raise TypeError
-        except TypeError:
-            raise OptionError('excclass option is not an exception class')
+        except TypeError as e:
+            raise OptionError(
+                'excclass option is not an exception class'
+            ) from e
```
Comment (birkenfeld) [inline pygments/filters/__init__.py:783]: should use `from None`

File: pygments/formatters/other.py [hunk 2]
```diff
@@ -74,16 +74,16 @@ def __init__(self, **options):
         if self.error_color is not None:
             try:
                 colorize(self.error_color, '')
-            except KeyError:
+            except KeyError as e:
                 raise ValueError("Invalid color %r specified" %
-                                 self.error_color)
+                                 self.error_color) from e
```
Comment (birkenfeld) [inline pygments/formatters/other.py:79]: `from None`

File: pygments/util.py [hunk 3]
```diff
@@ -72,14 +72,14 @@ def get_int_opt(options, optname, default=None):
     string = options.get(optname, default)
     try:
         return int(string)
-    except TypeError:
+    except TypeError as e:
         raise OptionError('Invalid type %r for option %s; you '
                           'must give an integer value' % (
-                              string, optname))
-    except ValueError:
+                              string, optname)) from e
```
Comment (birkenfeld) [inline pygments/util.py:78]: `from None`

File: pygments/util.py [hunk 4]
```diff
@@ -72,14 +72,14 @@ def get_int_opt(options, optname, default=None):
...
+    except ValueError as e:
         raise OptionError('Invalid value %r for option %s; you '
                           'must give an integer value' % (
-                              string, optname))
+                              string, optname)) from e
```
Comment (birkenfeld) [inline pygments/util.py:82]: `from None`

File: tests/test_basic_api.py [hunk 5]
```diff
@@ -66,10 +66,10 @@ def test_random_input(cls):
     inst = cls()
     try:
         tokens = list(inst.get_tokens(test_content))
-    except KeyboardInterrupt:
+    except KeyboardInterrupt as e:
         raise KeyboardInterrupt(
             'interrupted %s.get_tokens(): test_content=%r' %
-            (cls.__name__, test_content))
+            (cls.__name__, test_content)) from e
```
Comment (birkenfeld) [inline tests/test_basic_api.py:72]: `from None`

**Comment by cool-RR** [comment]
I'm strongly opposed to `from None`. I think that when people are debugging they have a very different mindset than when they are writing code. When you're debugging, and you get a very limited amount of information that may have been copy-pasted in an email by a non-technical person, you're very desperate for the information. Even if that information is a dead end for exploration, I at least want to know its content before I rule it out as a dead end.

Another issue is that some error-reporting systems show not only the traceback, but all the local variables on each of these levels. These supposedly irrelevant levels could have variables that make it clearer what this exception is about.

Obviously it's your project and your call, but I don't want to be the person responsible for another developer not getting exception information. I went over the places you marked and removed them from the PR so you could add these in a separate PR.

**Comment by cool-RR** [comment]
@birkenfeld Got a decision?

**Comment by cool-RR** [comment]
@birkenfeld Nudge

**Pull request closed without merging**